docs/spec/v1alpha1/gitrepositories.md

```diff
@@ -0,0 +1,216 @@
+# Git Repositories
+
+The `GitReposiory` API defines a source for artifacts coming from Git. 
+
+## Specification
+
+Git repository spec:
+
+```go
+// GitRepositorySpec defines the desired state of GitRepository
+type GitRepositorySpec struct {
+	// +kubebuilder:validation:Pattern="^(http|https|ssh)://"
+
+	// The repository URL, can be a HTTP or SSH address.
+	Url string `json:"url"`
+
+	// The secret name containing the Git credentials
+	// +optional
+	SecretRef *v1.LocalObjectReference `json:"secretRef,omitempty"`
+	
+	// The git reference to checkout and monitor for changes, defaults to master branch.
+	// +optional
+	Reference *GitRepositoryRef `json:"ref,omitempty"`
+
+	// The interval at which to check for repository updates.
+	Interval metav1.Duration `json:"interval"`
+}
+
+// GitRepositoryRef defines the git ref used for pull and checkout operations
+type GitRepositoryRef struct {
+	// The git branch to checkout, defaults to master.
+	// +optional
+	Branch string `json:"branch"`
+
+	// The git tag to checkout, takes precedence over branch.
+	// +optional
+	Tag string `json:"tag"`
+
+	// The git tag semver expression, takes precedence over tag.
+	// +optional
+	SemVer string `json:"semver"`
+
+	// The git commit sha to checkout, if specified branch and tag filters will be ignored.
+	// +optional
+	Commit string `json:"commit"`
+}
+```
+
+Git repository status:
+
+```go
+// GitRepositoryStatus defines the observed state of GitRepository
+type GitRepositoryStatus struct {
+	// +optional
+	Conditions []RepositoryCondition `json:"conditions,omitempty"`
+
+	// LastUpdateTime is the timestamp corresponding to the last status
+	// change of this repository.
+	// +optional
+	LastUpdateTime *metav1.Time `json:"lastUpdateTime,omitempty"`
+
+	// URI for the artifacts of the last successful repository sync.
+	// +optional
+	Artifacts string `json:"artifacts,omitempty"`
```

Comment:
For my (non design drafted) initial implementation of the `HelmRepository` index I changed this to `Artifact` (due to it being a single file). This creates a minimal divergence between both types that may be confusing.

Can we get away with naming this `Artifact` as it produces _an artifact from the Git repository_, or was the idea that this field represents the tarball with _artifacts taken from the Git repository_?

Reply:
Renamed to `Artifact`.